website/docs/class-of.md

```diff
@@ -1,75 +1,200 @@
 
-sig {params(x: T.class_of(M)).void}
-def test(x)
-  x.foo
+class MyClass
+  include MyInterface
 end
+```
 
+This will make `some_instance_method` and `some_class_method` available on
+`MyClass`, just like before. But if we try to replace `T.class_of(MyClass)` with
+`T.class_of(MyInterface)`, it doesn’t work:
 
-M.foo                      # ok
-M.is_a?(M.singleton_class) # => true
-test(M)                    # ok, given above
+```ruby
+sig {params(x: T.class_of(MyInterface)).void}  # ← sig has changed
+def example2(x)
+  x.new.some_instance_method  # error: `new` does not exist
+  x.some_class_method         # error: `some_class_method` does not exist
+end
 
-C.foo                      # not ok: this raises a NoMethodError when run
-C.is_a?(M.singleton_class) # => false
-test(C)                    # not ok, given above
+example2(MyClass)             # error: Expected `T.class_of(MyInterface)`
+                              #        but found `T.class_of(MyClass)`
 ```
 
-<a href="https://sorbet.run/#%23%20typed%3A%20true%0A%0Aextend%20T%3A%3ASig%0A%0Amodule%20M%0A%20%20def%20self.foo%3B%20end%0Aend%0Aclass%20C%3B%20include%20M%3B%20end%0A%0Asig%20%7Bparams(x%3A%20T.class_of(M)).void%7D%0Adef%20test(x)%0A%20%20x.foo%0Aend%0A%0A%0AM.foo%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%23%20ok%0AM.is_a%3F(M.singleton_class)%20%23%20%3D%3E%20true%0Atest(M)%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%23%20ok%2C%20given%20above%0A%0AC.foo%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%23%20not%20ok%3A%20this%20raises%20a%20NoMethodError%20when%20run%0AC.is_a%3F(M.singleton_class)%20%23%20%3D%3E%20false%0Atest(C)%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%23%20not%20ok%2C%20given%20above">
-  → View on sorbet.run
-</a>
+**These errors are correct**, and we can verify them in the Ruby REPL. First,
+let's explain the error on the last line above:
+
+```
+❯ MyClass.singleton_class.ancestors
+=> [#<Class:MyClass>, MyInterface::ClassMethods, #<Class:Object>, T::Private::Methods::MethodHooks, #<Class:BasicObject>, Class, Module, T::Sig, Object, Kernel, BasicObject]
+```
+
+The first two ancestors of the `MyClass` object are itself and
+`MyInterface::ClassMethods`. But notably, `#<Class:MyInterface>` **does not**
+appear in this list, so Sorbet is correct to say that `MyClass` does not have
+type `T.class_of(MyInterface)`. This is because neither `include` nor `extend`
+in Ruby will cause `#<Class:MyInterface>` to appear in any ancestors list.
+
+Next, let's explain the other two errors:
+
+```
+❯ MyInterface.singleton_class.ancestors
+=> [#<Class:MyInterface>, T::Private::MixesInClassMethods, T::Helpers, Module, T::Sig, Object, Kernel, BasicObject]
+```
+
+For the `MyInterface` class object, we see that its only ancestor is itself
+(ignoring those common ancestors). Notably, **none** of the classes in this list
+define either a method called `new` (because `Class` is not there) nor
```

Comment:
```suggestion
defines either a method called `new` (because `Class` is not there) nor
```

Reply:
I think this should be `define` (plural) from both my intuition of "does this sound right" and my reading of a handful of grammar articles online.

Reply:
Works for me!